### PR TITLE
Dev/architecture aa test report

### PR DESCRIPTION
**Pull request by Dmatryus**
* Report for AATest.
* AATestWithStratification

### CODE REVIEW

File: hypex/analyzers/aa2.py [hunk 1]
```diff
@@ -37,40 +39,71 @@ def _analyze_aa_score(
...
         self.key = "aa score"
         return self._set_value(data, result)
 
-    def _analyze_best_split(
+    def build_splitter_from_id(self, splitter_id: str):
+        return self.AA_SPLITER_CLASS_MAPPING.get(
+            splitter_id[: splitter_id.find(ID_SPLIT_SYMBOL)]
+        ).build_from_id(splitter_id)
+
+    def _geet_best_split(
```
Comment (TonyKatkov89) [inline hypex/analyzers/aa2.py:52]: maybe "_get_best_split"?

File: hypex/executor/executor.py [hunk 2]
```diff
@@ -62,6 +62,19 @@ def set_params(self, params: SetParamsDictTypes) -> None:
             )
         self._generate_id()
 
+    def init_from_hash(self, hash: str) -> None:
```
Comment (VasilievaKA) [inline hypex/executor/executor.py:65]: ~-> None~

File: hypex/experiments/base.py [hunk 3]
```diff
@@ -76,11 +76,11 @@ class OnRoleExperiment(Experiment):
     def __init__(
         self,
         executors: List[Executor],
-        role: ABCRole,
+        role: Union[ABCRole, Sequence[ABCRole]],
         transformer: Optional[bool] = None,
         key: Any = "",
     ):
-        self.role: ABCRole = role
+        self.role: List[ABCRole] = [role] if isinstance(role, ABCRole) else list(role)
```
Comment (VasilievaKA) [inline hypex/experiments/base.py:83]: use `Adapter.to_list(role)`

File: hypex/reporters/aa.py [hunk 4]
```diff
@@ -53,31 +57,29 @@ def _convert_struct_dict_to_dataset(data: Dict) -> Dataset:
         result = []
         for feature, groups in data.items():
```
Comment (VasilievaKA) [inline hypex/reporters/aa.py:58]: why in typing u use `data: Dict[str, bool]`, but in cicle variable `groups` may have items?

File: hypex/reporters/aa.py [hunk 5]
```diff
@@ -53,31 +57,29 @@ def _convert_struct_dict_to_dataset(data: Dict) -> Dataset:
...
+        struct_dict = OneAADictReporter._get_struct_dict(data)
+        return OneAADictReporter._convert_struct_dict_to_dataset(struct_dict)
 
     @staticmethod
     def get_splitter_id(data: ExperimentData):
-        return data.get_one_id(AASplitter, ExperimentDataEnum.additional_fields)
+        for c in [AASplitter, AASplitterWithStratification]:
+            try:
+                return data.get_one_id(c, ExperimentDataEnum.additional_fields)
+            except NotFoundInExperimentDataError:
```
Comment (VasilievaKA) [inline hypex/reporters/aa.py:81]: try-except?)

**Review by VasilievaKA — APPROVED** [review]
🧙‍♂️🌭

File: hypex/splitters/aa.py [hunk 6]
```diff
@@ -72,23 +92,39 @@ def execute(self, data: ExperimentData) -> ExperimentData:
...
-#             result = group_ds if result is None else result.append(group_ds)
-#         return result["group"]
-#
+class AASplitterWithStratification(AASplitter):
+    @staticmethod
+    def _inner_function(
+        data: Dataset,
+        random_state: Optional[int] = None,
+        control_size: float = 0.5,
+        grouping_fields=None,
```
Comment (VasilievaKA) [inline hypex/splitters/aa.py:101]: add type to grouping_field

**Pull request merged**